### PR TITLE
Rename Core >safe core and origin core >mosaic core

### LINKED ISSUE #500: Rename Core to SafeCore and OriginCore to MosaicCore
Rename Core to SafeCore and OriginCore to MosaicCore.
Move the SafeCore to Core directory.

### PR DESCRIPTION
**Pull request by sarvesh-ost**
Fixes #500  

Renamed `core` to `safe core` and `origin core` to `mosaic core`.

### CODE REVIEW

**Comment by schemar** [comment]
@sarvesh-ost can you please resolve the conflicts before we review it?

**Comment by sarvesh-ost** [comment]
> @sarvesh-ost can you please resolve the conflicts before we review it?

It's not available for review.

File: contracts/core/MosaicCore.sol [hunk 1]
```diff
@@ -15,19 +15,19 @@ pragma solidity ^0.5.0;
...
 /**
- * @title OriginCore is a meta-blockchain with staked validators on Ethereum.
+ * @title MosaicCore is a meta-blockchain with staked validators on Ethereum.
  */
-contract OriginCore is
-    OriginCoreInterface,
-    OriginCoreConfig,
+contract MosaicCore is
+MosaicCoreInterface,
+MosaicCoreConfig,
```
Comment (schemar) [inline contracts/core/MosaicCore.sol]: Maybe indent? Not sure what the official rules are.

File: contracts/core/PollingPlace.sol [hunk 2]
```diff
@@ -233,10 +233,10 @@ contract PollingPlace is PollingPlaceInterface {
         BlockStoreInterface originBlockStore = BlockStoreInterface(_originBlockStore);
         BlockStoreInterface auxiliaryBlockStore = BlockStoreInterface(_auxiliaryBlockStore);
 
-        originCoreIdentifier = originBlockStore.getCoreIdentifier();
+        mosaicCoreIdentifier = originBlockStore.getCoreIdentifier();
         auxiliaryCoreIdentifier = auxiliaryBlockStore.getCoreIdentifier();
 
-        blockStores[originCoreIdentifier] = originBlockStore;
+        blockStores[mosaicCoreIdentifier] = originBlockStore;
```
Comment (schemar) [inline contracts/core/PollingPlace.sol]: This should stay as origin core identifier. This is about the core identifiers on origin and auxiliary, not about "origin core". So `PollingPlace` should not change.

Comment (sarvesh-ost) [reply contracts/core/PollingPlace.sol]: Done

File: contracts/core/MosaicCore.sol (hunk 1 above)
Comment (sarvesh-ost) [reply contracts/core/MosaicCore.sol]: Done

File: contracts/test/core/MockCore.sol [hunk 3]
```diff
@@ -1,14 +1,14 @@
...
-import "../../gateway/Core.sol";
+import "../../gateway/SafeCore.sol";
 
 /**
  * @title CoreMock contract
  *
  * @notice Used for test only
  */
-contract MockCore is Core {
+contract MockCore is SafeCore {
```
Comment (schemar) [inline contracts/test/core/MockCore.sol]: `MockSafeCore`?

File: test/core/kernel_gateway/activate_kernel.js [hunk 4]
```diff
@@ -142,7 +142,7 @@ contract('KernelGateway.activateKernel()', async (accounts) => {
 
     assert.strictEqual(
       eventData._originCoreIdentifier,
-      originCoreIdentifier,
+        originCoreIdentifier,
```
Comment (schemar) [inline test/core/kernel_gateway/activate_kernel.js]: ?

File: test/core/origin_core/constructor.js [hunk 5]
```diff
@@ -24,12 +24,12 @@ const web3 = require('../../test_lib/web3.js');
 const BN = require('bn.js');
 const Utils = require('../../test_lib/utils.js');
 
-const OriginCore = artifacts.require('OriginCore');
+const MosaicCore = artifacts.require('MosaicCore');
```
Comment (schemar) [inline test/core/origin_core/constructor.js]: also rename the directory of these tests

**Pull request merged**